realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -668,6 +668,52 @@ public static AllTypes copy(Realm realm, AllTypes newObject, boolean update, Map
         return realmObject;
     }
 
+    public static AllTypes createDetachedCopy(AllTypes realmObject, int currentDepth, int maxDepth, Map<RealmObject, CacheData<RealmObject>> cache) {
+        if (currentDepth > maxDepth || realmObject == null) {
+            return null;
+        }
+        CacheData<AllTypes> cachedObject = (CacheData) cache.get(realmObject);
+        AllTypes standaloneObject;
+        if (cachedObject != null) {
+            // Reuse cached object or recreate it because it was encountered at a lower depth.
+            if (currentDepth >= cachedObject.minDepth) {
+                return cachedObject.object;
+            } else {
+                standaloneObject = cachedObject.object;
+                cachedObject.minDepth = currentDepth;
+            }
+        } else {
+            standaloneObject = new AllTypes();
+            cache.put(realmObject, new RealmObjectProxy.CacheData<RealmObject>(currentDepth, standaloneObject));
+        }
+        standaloneObject.setColumnString(realmObject.getColumnString());
+        standaloneObject.setColumnLong(realmObject.getColumnLong());
+        standaloneObject.setColumnFloat(realmObject.getColumnFloat());
+        standaloneObject.setColumnDouble(realmObject.getColumnDouble());
+        standaloneObject.setColumnBoolean(realmObject.isColumnBoolean());
+        standaloneObject.setColumnDate(realmObject.getColumnDate());
+        standaloneObject.setColumnBinary(realmObject.getColumnBinary());
+
+        // Deep copy of columnObject
+        standaloneObject.setColumnObject(AllTypesRealmProxy.createDetachedCopy(realmObject.getColumnObject(), currentDepth + 1, maxDepth, cache));
+
+        // Deep copy of columnRealmList
+        if (currentDepth == maxDepth) {
+            standaloneObject.setColumnRealmList(null);
```

Comment:
This will be a little bit inconsistency since the `RealmList` of `RealmObjectProxy` cannot be null. But I don't know if there is anything better we can do except making the `RealmList` nullable.